src/logic/div.rs

```diff
@@ -0,0 +1,1290 @@
+                    div_sd,
+                    div_sb,
+                    div_lz
+                );
+                return true
+            },
+            None => return false,
+        }
+    }
+
+    /// This function is intended to be inlined into all of the unsigned quotient and remainder
+    /// functions for optimal assembly.
+    /// `duo` is divided by `div`, and the remainder is assigned to `duo` and quotient assigned
+    /// to `div`
+    /// `false` is returned if division by zero happened. Nothing is modified in the case of
+    /// division by zero.
+    #[inline]
+    pub(crate) fn aarons_algorithm_remdiv(duo: &mut [Digit], div: &mut [Digit]) -> bool {
+        if ApInt::aarons_algorithm_divrem(duo, div) {
+            let mut temp;
+            for i in 0..duo.len() {
+                temp = duo[i];
+                duo[i] = div[i];
+                div[i] = temp;
+            }
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Divides `lhs` by `rhs` using **unsigned** interpretation and sets `lhs` equal to the
+    /// quotient and `rhs` equal to the remainder.
+    pub fn wrapping_udivrem_assign(lhs: &mut ApInt, rhs: &mut ApInt) -> Result<()> {
+        match ApInt::zip_access_data_mut_both(lhs, rhs)? {
+            ZipDataAccessMutBoth::Inl(duo,div) => {
+                if *div != Digit::zero() {
+                    let temp = duo.wrapping_divrem(*div);
+                    *duo = temp.0;
+                    *div = temp.1;
+                    return Ok(())
+                }
+            }
+            ZipDataAccessMutBoth::Ext(duo,div) => {
+                if ApInt::aarons_algorithm_divrem(duo, div) {
+                    return Ok(())
+                }
+            }
+        }
+        //Note that the typical places `Err` `Ok` are returned is switched. This is because
+        //`rhs.is_zero()` is found as part of finding `duo_sd` inside `aarons_algorithm_divrem`,
+        //and `lhs.clone()` cannot be performed inside the match statement
+		return Err(Error::division_by_zero(DivOp::UnsignedDivRem, lhs.clone()))
+    }
+
+    /// Divides `lhs` by `rhs` using **unsigned** interpretation and sets `lhs` equal to the
+    /// remainder and `rhs` equal to the quotient.
+    pub fn wrapping_uremdiv_assign(lhs: &mut ApInt, rhs: &mut ApInt) -> Result<()> {
+        match ApInt::zip_access_data_mut_both(lhs, rhs)? {
+            ZipDataAccessMutBoth::Inl(duo,div) => {
+                if *div != Digit::zero() {
+                    let temp = duo.wrapping_divrem(*div);
+                    *duo = temp.1;
+                    *div = temp.0;
+                    return Ok(())
+                }
+            }
+            ZipDataAccessMutBoth::Ext(duo,div) => {
+                if ApInt::aarons_algorithm_remdiv(duo, div) {
+                    return Ok(())
+                }
+            }
+        }
+		return Err(Error::division_by_zero(DivOp::UnsignedRemDiv, lhs.clone()))
```

Comment:
strange indentation

Reply:
That is bizarre there were tabs instead of spaces there. I don't know how those survived but they are gone now